samcli/lib/build/app_builder.py

```diff
@@ -180,7 +180,13 @@ def _build_function(self, function_name, codeuri, runtime):
             # By default prefer to build in-process for speed
             build_method = self._build_function_in_process
             if self._container_manager:
-                build_method = self._build_function_on_container
+
+                container_build_supported, reason = supports_build_in_container(config)
+                if container_build_supported:
+                    build_method = self._build_function_on_container
+                else:
+                    LOG.warning(reason)
```

Comment:
Can we color the output? Or maybe add a "WARNING" to the beginning of the output (since we don't have a way to turn off colors)? I ask because this warning gets lost in the output, unless you are looking for it.

```
2019-04-03 09:00:19 Starting Build inside a container
2019-04-03 09:00:19 Building resource 'HelloWorldFunction'
2019-04-03 09:00:19 We do not support building dotnet Lambda functions within a container. Most dotnet functions can be successfully built without a container.
2019-04-03 09:00:19 Continuing build without a container
2019-04-03 09:00:19 Running DotnetCliPackageBuilder:GlobalToolInstall
2019-04-03 09:00:20
2019-04-03 09:00:23 Tool 'amazon.lambda.tools' was reinstalled with the latest stable version (version '3.2.0').
2019-04-03 09:00:23 Running DotnetCliPackageBuilder:RunPackageAction
2019-04-03 09:00:29 Amazon Lambda Tools for .NET Core applications (3.2.0)
Project Home: https://github.com/aws/aws-extensions-for-dotnet-cli, https://github.com/aws/aws-lambda-dotnet

Executing publish command
Deleted previous publish folder
... invoking 'dotnet publish', working folder '/Users/jfuss/sam-github/aws-sam-cli/sam-app/src/HelloWorld/bin/Debug/netcoreapp2.1/publish'
... Disabling compilation context to reduce package size. If compilation context is needed pass in the "/p:PreserveCompilationContext=false" switch.
... publish: Microsoft (R) Build Engine version 15.9.20+g88f5fadfbe for .NET Core
... publish: Copyright (C) Microsoft Corporation. All rights reserved.
... publish:   Restore completed in 55.32 ms for /Users/jfuss/sam-github/aws-sam-cli/sam-app/src/HelloWorld/HelloWorld.csproj.
... publish:   HelloWorld -> /Users/jfuss/sam-github/aws-sam-cli/sam-app/src/HelloWorld/bin/Debug/netcoreapp2.1/rhel.7.2-x64/HelloWorld.dll
... publish:   HelloWorld -> /Users/jfuss/sam-github/aws-sam-cli/sam-app/src/HelloWorld/bin/Debug/netcoreapp2.1/publish/
Changed permissions on published file (chmod +rx Amazon.Lambda.APIGatewayEvents.dll).
Changed permissions on published file (chmod +rx Amazon.Lambda.Core.dll).
Changed permissions on published file (chmod +rx Amazon.Lambda.Serialization.Json.dll).
Changed permissions on published file (chmod +rx HelloWorld.deps.json).
Changed permissions on published file (chmod +rx HelloWorld.dll).
Changed permissions on published file (chmod +rx HelloWorld.pdb).
Changed permissions on published file (chmod +rx HelloWorld.runtimeconfig.json).
Changed permissions on published file (chmod +rx Newtonsoft.Json.dll).
Zipping publish folder /Users/jfuss/sam-github/aws-sam-cli/sam-app/src/HelloWorld/bin/Debug/netcoreapp2.1/publish to /Users/jfuss/sam-github/aws-sam-cli/sam-app/.aws-sam/build/HelloWorldFunction/HelloWorld.zip
Creating directory /Users/jfuss/sam-github/aws-sam-cli/sam-app/.aws-sam/build/HelloWorldFunction
... zipping:   adding: Amazon.Lambda.APIGatewayEvents.dll (deflated 61%)
... zipping:   adding: Amazon.Lambda.Core.dll (deflated 57%)
... zipping:   adding: Amazon.Lambda.Serialization.Json.dll (deflated 56%)
... zipping:   adding: HelloWorld.deps.json (deflated 76%)
... zipping:   adding: HelloWorld.dll (deflated 54%)
... zipping:   adding: HelloWorld.pdb (deflated 28%)
... zipping:   adding: HelloWorld.runtimeconfig.json (deflated 23%)
... zipping:   adding: Newtonsoft.Json.dll (deflated 63%)
Created publish archive (/Users/jfuss/sam-github/aws-sam-cli/sam-app/.aws-sam/build/HelloWorldFunction/HelloWorld.zip).
Lambda project successfully packaged: /Users/jfuss/sam-github/aws-sam-cli/sam-app/.aws-sam/build/HelloWorldFunction/HelloWorld.zip

Build Succeeded

Built Artifacts  : .aws-sam/build
Built Template   : .aws-sam/build/template.yaml

Commands you can use next
=========================
[*] Invoke Function: sam local invoke
[*] Package: sam package --s3-bucket <yourbucket>
```

Reply:
Agreed. I will find a different mechanism